homeassistant/components/sensor/__init__.py

```diff
@@ -747,13 +747,13 @@ def state(self) -> Any:
 
         return value
 
-    def _suggested_precision_or_none(self) -> int | None:
-        """Return suggested display precision, or None if not set."""
+    def _display_precision_or_none(self) -> int | None:
+        """Return display precision, or None if not set."""
         assert self.registry_entry
-        if (sensor_options := self.registry_entry.options.get(DOMAIN)) and (
-            precision := sensor_options.get("suggested_display_precision")
-        ) is not None:
-            return cast(int, precision)
+        sensor_options = self.registry_entry.options.get(DOMAIN)
+        for option in ("display_precision", "suggested_display_precision"):
+            if sensor_options and (precision := sensor_options.get(option)) is not None:
+                return cast(int, precision)
```

Comment:
Would be slightly cheaper to not iterate `if not (sensor_options := sensor_options = self.registry_entry.options.get(DOMAIN)):`